discretisedfield/plotting/mpl.py

```diff
@@ -0,0 +1,873 @@
+import mpl_toolkits.axes_grid1.inset_locator
+import matplotlib.pyplot as plt
+import numpy as np
+
+import discretisedfield.util as dfu
+import ubermagutil.units as uu
+
+
+class Mpl:
+    """Matplotlib-based plotting convenience methods.
+
+    Parameters
+    ----------
+    data : df.Field
+
+        Field sliced with a plane, e.g. field.plane('x').
+
+    """
+
+    def __init__(self, data):
+        # We never allow plotting fields that are not sliced.
+        if not data.mesh.attributes['isplane']:
+            msg = 'The field must be sliced before it can be plotted.'
+            raise ValueError(msg)
+
+        self.data = data  # TODO: Consider renaming data to field.
+        self.planeaxis = dfu.raxesdict[data.mesh.attributes['planeaxis']]
+
+
+    # TODO: I assume the functionality from plot is being moved to plot_simplified?
+    def plot_simplified(self,
+                        ax=None,
+                        figsize=None,
+                        multiplier=None,
+                        scalar_args={},
+                        vector_args={},
```

Comment:
`scalar_args` and `vector_args` must be `None` because of some Python specialities. The default arguments are only evaluated once and reused for later calls. If e.g. `scalar_args` is modified during the first execution these changes will persist for later calls. Therefore, default arguments must generally be immutable (no `list`, `dict`, sometimes `tuple`, ...) unless this very special functionality is desired.

Reply:
Fixed in cad1126